compiler/passes/errorHandling.cpp

```diff
@@ -420,9 +420,26 @@ bool ErrorHandlingVisitor::enterCallExpr(CallExpr* node) {
         VarSymbol* handledError = toVarSymbol(catchFilter->sym);
 
         VarSymbol* throwingHandledVar = newTemp("throwingHandledError", dtBool);
-        CallExpr*  throwingHandled    = new CallExpr(PRIM_EQUAL, thrownError, handledError);
-
         throwBlock->insertAtTail(new DefExpr(throwingHandledVar));
+
+        CallExpr* throwingHandled = NULL;
+
+        if (handledError->type != dtError) {
+          // We need to cast these to the same base type before comparing using
+          // pointer comparison if the type is a subtype of error
```

Comment:
Is other code ensuring that `handledError` is a subtype of `Error`? Maybe add a test of this?

Reply:
It seems as though that is contingent on being able to throw something that is not a subtype of Error.  If you can't throw it, the catch block never triggers.  Trying to throw something that isn't a subtype of Error gives an error message, but not the most helpful one:
```
notAnErrorSubtype-thrown.chpl:5: In function 'main':
notAnErrorSubtype-thrown.chpl:7: error: unresolved call 'chpl_fix_thrown_error(unmanaged Foo)'
$CHPL_HOME/modules/internal/ChapelError.chpl:339: note: candidates are: chpl_fix_thrown_error(err: borrowed Error) [162960]
$CHPL_HOME/modules/internal/ChapelError.chpl:345: note:                 chpl_fix_thrown_error(type errType) [162981]
$CHPL_HOME/modules/internal/ChapelError.chpl:352: note:                 chpl_fix_thrown_error(err: unmanaged Error) [163059]
$CHPL_HOME/modules/internal/ChapelError.chpl:367: note:                 chpl_fix_thrown_error(ref err: owned Error) [163134]
```

Reply:
I wrote five variant tests, I'll commit them (some as futures) in this PR and open a separate issue for the better error message when throwing something that shouldn't be thrown

Reply:
Thanks!